inference-engine/tests/functional/plugin/cpu/single_layer_tests/batch_to_space.cpp

```diff
@@ -0,0 +1,142 @@
+
+namespace CPULayerTestsDefinitions  {
+
+typedef std::tuple<
+        LayerTestsDefinitions::batchToSpaceParamsTuple,
+        CPUSpecificParams> BatchToSpaceLayerCPUTestParamSet;
+
+class BatchToSpaceCPULayerTest : public testing::WithParamInterface<BatchToSpaceLayerCPUTestParamSet>,
+                                 virtual public LayerTestsUtils::LayerTestsCommon, public CPUTestsBase {
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<BatchToSpaceLayerCPUTestParamSet> &obj) {
+        LayerTestsDefinitions::batchToSpaceParamsTuple basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = obj.param;
+        std::ostringstream result;
+        result << LayerTestsDefinitions::BatchToSpaceLayerTest::getTestCaseName(
+                testing::TestParamInfo<LayerTestsDefinitions::batchToSpaceParamsTuple>(basicParamsSet, 0));
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+        return result.str();
+    }
+
+protected:
+    void SetUp() override {
+        LayerTestsDefinitions::batchToSpaceParamsTuple basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+        std::vector<size_t> inputShape;
+        std::vector<int64_t> blockShape, cropsBegin, cropsEnd;
+        InferenceEngine::Precision netPrecision;
+        std::tie(blockShape, cropsBegin, cropsEnd, inputShape, netPrecision, inPrc, outPrc, inLayout, outLayout, targetDevice) = basicParamsSet;
+        inPrc = outPrc = netPrecision;
+
+        if (strcmp(netPrecision.name(), "U8") == 0)
+            selectedType = std::string("unknown_") + "I8";
+        else
+            selectedType = std::string("unknown_") + netPrecision.name();
+
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
+        auto paramOuts = ngraph::helpers::convert2OutputVector(
+                ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+        auto b2s = ngraph::builder::makeBatchToSpace(paramOuts[0], ngPrc, blockShape, cropsBegin, cropsEnd);
+        b2s->get_rt_info() = getCPUInfo();
+        ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(b2s)};
+        function = std::make_shared<ngraph::Function>(results, params, "BatchToSpace");
+    }
+};
+
+TEST_P(BatchToSpaceCPULayerTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    Run();
+    CheckPluginRelatedResults(executableNetwork, "BatchToSpace");
+};
+
+namespace {
+
+const std::vector<Precision> precisions = {
+        Precision::U8,
+        Precision::I8,
+        Precision::I32,
+        Precision::FP32,
+        Precision::BF16
+};
+
+const std::vector<std::vector<int64_t>> blockShape4D  = {{1, 1, 1, 2}, {1, 2, 2, 1}, {1, 1, 2, 2}, {1, 2, 1, 2}};
```

Comment:
I think we have to cover blocks equals not only 1 and 2. For example it sis worthwhile to cover the case from the spec: https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/movement/BatchToSpace_2.md